src/hidimstat/knockoffs.py

```diff
@@ -1,109 +1,570 @@
 from sklearn.preprocessing import StandardScaler
-from sklearn.utils.validation import check_memory
+from sklearn.covariance import LedoitWolf
+from sklearn.linear_model import LassoCV
+from sklearn.model_selection import KFold
+from sklearn.utils import check_random_state
+from joblib import Parallel, delayed
 
-from .gaussian_knockoff import _estimate_distribution, gaussian_knockoff_generation
-from .stat_coef_diff import _coef_diff_threshold, stat_coef_diff
+from hidimstat.gaussian_knockoff import (
+    gaussian_knockoff_generation,
+    repeat_gaussian_knockoff_generation,
+)
+from hidimstat.utils import fdr_threshold, quantile_aggregation
+
+
+def preconfigure_estimator_LassoCV(estimator, X, X_tilde, y, n_alphas=10):
+    """
+    Configure the estimator for Model-X knockoffs.
+
+    This function sets up the regularization path for the Lasso estimator
+    based on the input data and the number of alphas to use. The regularization
+    path is defined by a sequence of alpha values, which control the amount
+    of shrinkage applied to the coefficient estimates.
+
+    Parameters
+    ----------
+    estimator : sklearn.linear_model._base.LinearModel
+        An instance of a linear model estimator from sklearn.linear_model.
+        This estimator will be used to fit the data and compute the test
+        statistics. In this case, it should be an instance of LassoCV.
+
+    X : 2D ndarray (n_samples, n_features)
+        The original design matrix.
+
+    X_tilde : 2D ndarray (n_samples, n_features)
+        The knockoff design matrix.
+
+    y : 1D ndarray (n_samples, )
+        The target vector.
+
+    n_alphas : int, optional (default=10)
+        The number of alpha values to use to instantiate the cross-validation.
+
+    Returns
+    -------
+    None
+        The function modifies the `estimator` object in-place.
+
+    Raises
+    ------
+    TypeError
+        If the estimator is not an instance of LassoCV.
+
+    Notes
+    -----
+    This function is specifically designed for the Model-X knockoffs procedure,
+    which combines original and knockoff variables in the design matrix.
+    """
+    if type(estimator).__name__ != "LassoCV":
+        raise TypeError("You should not use this function for configure your estimator")
+
+    n_features = X.shape[1]
+    X_ko = np.column_stack([X, X_tilde])
+    alpha_max = np.max(np.dot(X_ko.T, y)) / (2 * n_features)
+    alphas = np.linspace(alpha_max * np.exp(-n_alphas), alpha_max, n_alphas)
+    estimator.alphas = alphas
 
 
 def model_x_knockoff(
     X,
     y,
-    fdr=0.1,
-    offset=1,
-    method="equi",
-    statistics="lasso_cv",
-    shrink=False,
+    estimator=LassoCV(
+        n_jobs=None,
+        verbose=0,
+        max_iter=200000,
+        cv=KFold(n_splits=5, shuffle=True, random_state=0),
+        tol=1e-6,
+    ),
+    preconfigure_estimator=preconfigure_estimator_LassoCV,
     centered=True,
-    cov_estimator="ledoit_wolf",
-    verbose=False,
-    memory=None,
```

Comment:
Why are you removing memory ? You're going to add it back in a forthcoming PR...

Reply:
In my initial plan, I removed all the memory optimisation for all the methods because I wanted to have a global vision of the code and the usage of it before to introduce some optimisations.
After the discussion in the PR #127, I changed my point of view.

Reply:
done